src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTest.scala

```diff
@@ -281,62 +281,84 @@ class E2EHyperspaceRulesTest extends QueryTest with HyperspaceSuite {
-          |LOCATION '$nonPartitionedDataPath'
-        """.stripMargin)
-
-      val leftDf = spark.table("t1")
-      val rightDf = spark.table("t2")
+  test("E2E test for join query on external catalog tables") {
+    Seq(true, false).map { useDDL =>
+      withIndex("leftIndex", "rightIndex") {
+        withTable("t1", "t2") {
+          if (useDDL) {
+            spark.sql("CREATE TABLE t1 (c1 string, c3 string) USING PARQUET " +
+              s"LOCATION '$nonPartitionedDataPath'")
+            spark.sql("CREATE TABLE t2 (c3 string, c4 int) USING PARQUET " +
+              s"LOCATION '$nonPartitionedDataPath'")
+          } else {
+            val table1Location = testDir + "tables/t1"
+            val table2Location = testDir + "tables/t2"
+            val originalDf = spark.read.parquet(nonPartitionedDataPath)
+            originalDf.select("c1", "c3").write.option("path", table1Location).saveAsTable("t1")
+            originalDf.select("c3", "c4").write.option("path", table2Location).saveAsTable("t2")
+          }
 
-      val leftDfIndexConfig = IndexConfig("leftIndex", Seq("c3"), Seq("c1"))
-      val rightDfIndexConfig = IndexConfig("rightIndex", Seq("c3"), Seq("c4"))
+          assert(spark.catalog.getTable("t1").tableType == "EXTERNAL")
+          assert(spark.catalog.getTable("t2").tableType == "EXTERNAL")
 
-      hyperspace.createIndex(leftDf, leftDfIndexConfig)
-      hyperspace.createIndex(rightDf, rightDfIndexConfig)
+          val leftDf = spark.table("t1")
+          val rightDf = spark.table("t2")
+          val leftDfIndexConfig = IndexConfig("leftIndex", Seq("c3"), Seq("c1"))
+          val rightDfIndexConfig = IndexConfig("rightIndex", Seq("c3"), Seq("c4"))
+          hyperspace.createIndex(leftDf, leftDfIndexConfig)
+          hyperspace.createIndex(rightDf, rightDfIndexConfig)
 
-      // Test whether indexes work with catalog tables or not
-      def query(): DataFrame = spark.sql("SELECT t1.c1, t2.c4 FROM t1, t2 WHERE t1.c3 = t2.c3")
-      verifyIndexUsage(
-        query,
-        getIndexFilesPath(leftDfIndexConfig.indexName) ++
-          getIndexFilesPath(rightDfIndexConfig.indexName))
+          // Test whether indexes work with catalog tables or not.
+          def query(): DataFrame =
+            spark.sql("SELECT t1.c1, t2.c4 FROM t1, t2 WHERE t1.c3 = t2.c3")
+
+          verifyIndexUsage(
+            query,
+            getIndexFilesPath(leftDfIndexConfig.indexName) ++
+              getIndexFilesPath(rightDfIndexConfig.indexName))
+        }
+      }
     }
   }
 
   test("E2E test for join query on managed catalog tables") {
-    withTable("t1", "t2") {
-      val table1Location = testDir + "tables/t1"
-      val table2Location = testDir + "tables/t2"
-      val originalDf = spark.read.parquet(nonPartitionedDataPath)
-      originalDf.select("c1", "c3").write.option("path", table1Location).saveAsTable("t1")
-      originalDf.select("c3", "c4").write.option("path", table2Location).saveAsTable("t2")
```

Comment:
If you pass "path" to the option, `saveAsTable` will create an external table.

Reply:
oh great so looks like the original tests had a bug. thanks!